### ISSUE TITLE
update Eigen to support CUDA 9.0

### ISSUE THREAD

**Issue by fwyzard** [cms-sw/cmsdist#3582]
Eigen should be compatible with CUDA 9.0 starting from [#c7dc0a897676](https://bitbucket.org/eigen/eigen/commits/c7dc0a89767617cb32cd9ab0c2aed9401db71fac) :

```hg
changeset:   10595:c7dc0a897676
parent:      10587:51c95eb5952b
user:        Benoit Steiner <benoit.steiner.goog@gmail.com>
date:        Thu Aug 31 02:49:39 2017 +0000
summary:     Added support for CUDA 9.0.
```

**Comment by fwyzard** [comment]
Fixed by #3587

**Comment by smuzaffar** [comment]
lets first test it in DEVEL IBs https://github.com/cms-sw/cmsdist/pull/3589
